### PR TITLE
@esteban/edit profiles to transactions

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-2721

## What changed (plus any additional context for devs)

Allow sending a multicall or setText transaction when editing profiles

## PoW (screenshots / screen recordings)

https://user-images.githubusercontent.com/12115171/156658938-332a66c2-51ee-414a-9ca6-b10966f7c14b.mov

## Dev checklist for QA: what to test

txs well formatted
records edition 

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels?

### CODE REVIEW

File: src/screens/ENSConfirmRegisterSheet.js [hunk 1]
```diff
@@ -63,11 +63,7 @@ export default function ENSConfirmRegisterSheet() {
     yearsDuration: duration,
   });
   const prevStepGasLimit = usePrevious(stepGasLimit);
-
-  const { blurFields, values } = useENSProfileForm();
-  useFocusEffect(() => {
-    blurFields();
```
Comment (jxom) [inline src/screens/ENSConfirmRegisterSheet.js:69]: you sure we don't need this?

File: src/screens/ENSConfirmRegisterSheet.js [hunk 2]
```diff
@@ -63,11 +63,7 @@ export default function ENSConfirmRegisterSheet() {
     yearsDuration: duration,
   });
   const prevStepGasLimit = usePrevious(stepGasLimit);
-
-  const { blurFields, values } = useENSProfileForm();
-  useFocusEffect(() => {
```
Comment (estebanmino) [inline src/screens/ENSConfirmRegisterSheet.js:68]: @jxom this was causing some issues on the records update, I would open this sheet and the records were being erased, am I missing something?

File: src/hooks/useENSProfileForm.ts [hunk 3]
```diff
@@ -98,7 +97,7 @@ export default function useENSProfileForm({
 
   // Set initial records in redux depending on user input (defaultFields)
   useEffect(() => {
-    if (defaultFields && isEmpty(defaultRecords)) {
+    if (defaultFields && !isEmpty(defaultRecords)) {
```
Comment (jxom) [inline src/hooks/useENSProfileForm.ts:100]: don't we want to fill the defaults when the records are empty?

File: src/screens/ENSConfirmRegisterSheet.js (hunk 2 above)
Comment (jxom) [reply src/screens/ENSConfirmRegisterSheet.js:68]: hmm, it shouldn't do that. might have been a merge issue upstream... i can have a look

File: src/hooks/useENSProfileForm.ts (hunk 3 above)
Comment (estebanmino) [reply src/hooks/useENSProfileForm.ts]: just changed it!

Comment (estebanmino) [reply src/hooks/useENSProfileForm.ts:101]: reverted back per video sent via slack

Comment (jxom) [reply src/hooks/useENSProfileForm.ts:101]: are we still reverting this?

File: src/screens/ENSConfirmRegisterSheet.js [hunk 4]
```diff
@@ -101,6 +101,8 @@ export default function ENSConfirmRegisterSheet() {
 
   useEffect(() => startPollingGasFees(), [startPollingGasFees]);
 
+  useEffect(() => blurFields());
```
Comment (jxom) [inline src/screens/ENSConfirmRegisterSheet.js:104]: What does this do? It doesn't have a dependency array, so it would be invoked on every render.

**Review by jxom — APPROVED** [review]

**Comment by jxom** [comment]
get it in!!

**Pull request merged**